apps/test/solutions/applab/ec_uicontrols.js

```diff
@@ -189,5 +193,98 @@ module.exports = {
+
+          Applab.onPuzzleComplete();
+        });
+      },
+      customValidator: function (assert) {
+        // No errors in output console
+        var debugOutput = document.getElementById('debug-output');
+        assert.equal(debugOutput.textContent, "");
+        return true;
+      },
+      expected: {
+        result: true,
+        testResult: TestResults.FREE_PLAY
+      },
+    },
+    {
+      description: "substantial changes in html sanitization do cause warnings",
+      timeout: 20000,
+      editCode: true,
+      xml: '' +
+      'write(\'' +
+        '<div id="container">' +
+          '<img id="image3" src="">' +
+          '<img id="image4" src="' + imageUrl + '" />' +
+          '<img id="image5" src="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw==" />' +
+          '<img id="image6" src="javascript:alert()" />' +
+          '<script>alert()</script>' +
+        '</div>' +
+      '\')\n;',
+
+      runBeforeClick: function (assert) {
+        // add a completion on timeout since this is a freeplay level
+        testUtils.runOnAppTick(Applab, 2, function () {
+          var expectedHtml = '' +
+            '<div id="container">' +
+            '<img id="image3" src="">' +
+            '<img id="image4" src="' + imageUrl + '">' +
+            '<img id="image5" src="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw==">' +
+            '<img id="image6">' +
+            '</div>';
+          assert.equal($('#divApplab #container')[0].outerHTML, expectedHtml, 'container has unexpected outerHTML');
+
+          Applab.onPuzzleComplete();
+        });
+      },
+      customValidator: function (assert) {
+        var expectedOutput ='' +
+          'WARNING: Line: 1: The following lines of HTML were modified or removed:\n' +
+          '<img id="image6" src="javascript:alert()" />\n' +
+          '<script>alert()\n</script>\n' +
+          'original html:\n' +
+          '<div id="container">' +
+          '<img id="image3" src="">' +
+          '<img id="image4" src="//localhost:8001/apps/static/flappy_promo.png" />' +
+          '<img id="image5" src="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw==" />' +
+          '<img id="image6" src="javascript:alert()" /><script>alert()</script></div>\n' +
+          'modified html:\n' +
+          '<div id="container">' +
+          '<img id="image3" src />' +
+          '<img id="image4" src="//localhost:8001/apps/static/flappy_promo.png" />' +
+          '<img id="image5" src="data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw==" />' +
+          '<img id="image6" /></div>';
+        var debugOutput = document.getElementById('debug-output');
+        assert.equal(debugOutput.textContent, expectedOutput);
+        return true;
+      },
+      expected: {
+        result: true,
+        testResult: TestResults.FREE_PLAY
+      },
+    }
```

Comment:
proposal: disable console.log on applab debug console errors in unit tests. https://www.pivotaltracker.com/n/projects/1256654